src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -774,12 +774,18 @@ def list_vm_ip_addresses(cmd, resource_group_name=None, vm_name=None):
                 network_info['privateIpAddresses'].append(ip_configuration.private_ip_address)
                 if ip_configuration.public_ip_address:
                     public_ip_address = ip_address_lookup[ip_configuration.public_ip_address.id]
-                    network_info['publicIpAddresses'].append({
+
+                    public_ip_addr_info = {
                         'id': public_ip_address.id,
                         'name': public_ip_address.name,
                         'ipAddress': public_ip_address.ip_address,
                         'ipAllocationMethod': public_ip_address.public_ip_allocation_method
-                    })
+                    }
+
+                    if hasattr(public_ip_address, 'zones'):
+                        public_ip_addr_info['zone'] = public_ip_address.zones[0] if public_ip_address.zones else None
```

Comment:
Alternative:
```
try:
    public_ip_addr_info['zone'] = public_ip_address.zones[0]
except (AttributeError, IndexError):
    pass
```

__Ask forgiveness, not permission.__ And avoids two `if` check.